### PR TITLE
Fixes for 1.4-M2 & HMPP

### PR DESCRIPTION
**Pull request by h0tk3y**

### CODE REVIEW

**Comment by qwwdfsad** [comment]
Also, please revert changes in `DebugProbesKt.bin` and mute the test [here](https://github.com/Kotlin/kotlinx.coroutines/blob/master/build.gradle#L185) instead

File: gradle/publish-bintray.gradle [hunk 1]
```diff
@@ -64,36 +64,37 @@ publishing {
...
         }
-
-        // disable metadata everywhere, but in native and js modules
-        if (type == 'maven' || type == 'metadata' || type == 'jvm') {
-            moduleDescriptorGenerator = null
-        }
     }
 }
 
+tasks.matching { it.name == "generatePomFileForKotlinMultiplatformPublication"}.configureEach {
```
[the remark this answers is not in the document]
Comment (h0tk3y) [reply gradle/publish-bintray.gradle]: It was redundant, I've removed it, thanks.

File: gradle.properties [hunk 2]
```diff
@@ -5,11 +5,11 @@
 # Kotlin
 version=1.3.6-SNAPSHOT
 group=org.jetbrains.kotlinx
-kotlin_version=1.3.71
+kotlin_version=1.4-M2-eap-72
```
[the remark this answers is not in the document]
Comment (h0tk3y) [reply gradle.properties]: Done, force-pushed the branch without this commit.

File: gradle.properties [hunk 3]
```diff
@@ -5,11 +5,11 @@
...
 
 # Dependencies
 junit_version=4.12
-atomicfu_version=0.14.2
+atomicfu_version=0.14.3-mpp-dev-10
```
[the remark this answers is not in the document]
Comment (h0tk3y) [reply gradle.properties]: Done, force-pushed the branch without this commit.

File: kotlinx-coroutines-core/build.gradle [hunk 4]
```diff
@@ -60,10 +60,10 @@ kotlin.sourceSets {
     }
 
     jvmTest.dependencies {
-        api "org.jetbrains.kotlinx:lincheck:$lincheck_version"
+        api("org.jetbrains.kotlinx:lincheck:$lincheck_version") { exclude(module: "kotlinx-coroutines-core") }
         api "org.jetbrains.kotlinx:kotlinx-knit-test:$knit_version"
         api "com.esotericsoftware:kryo:4.0.0"
-        implementation project (":android-unit-tests")
+        implementation(project.dependencies.project(path: ":android-unit-tests")) { exclude(module: "kotlinx-coroutines-core") }
```
[the remark this answers is not in the document]
Comment (h0tk3y) [reply kotlinx-coroutines-core/build.gradle]: Done, added it as a separate commit ccd1333

File: gradle/publish-bintray.gradle (hunk 1 above)
[the remark this answers is not in the document]
Comment (h0tk3y) [reply gradle/publish-bintray.gradle]: Same as https://github.com/Kotlin/kotlinx.coroutines/pull/2031#discussion_r426557295, see ccd1333.

File: gradle.properties (hunk 3 above)
[the remark this answers is not in the document]
Comment (h0tk3y) [reply gradle.properties]: Are you sure? The project won't build locally due to https://github.com/Kotlin/kotlinx.atomicfu/issues/131

**Pull request merged**